### PR TITLE
Improve description of dotnet restore --source

### PR DESCRIPTION
**Pull request by jiimaho**
## Summary

Tried to make the description very explicit about the fact that the nuget.config file won't be used. Please improve/change further if you want to.

Fixes #Issue_Number (if available)

### CODE REVIEW

**Comment by Thraka** [comment]
Does it actually ignore the whole file? There are a lot of other settings, such as the global packages folder, that I would assume it continues to use.

https://docs.microsoft.com/en-us/nuget/reference/nuget-config-file#example-config-file

@KathleenDollard please comment.

**Comment by jiimaho** [comment]
Well at least when I tried it doesn't use any of the specified sources in NuGet.Config if the `--source` parameter is provided. However as you say, i'm not sure about the other settings. 

Anyone else knows this?

File: docs/core/tools/dotnet-restore.md [hunk 1]
```diff
@@ -171,7 +171,7 @@ Specifies a runtime for the package restore. This is used to restore packages fo
 
 `-s|--source <SOURCE>`
 
-Specifies a NuGet package source to use during the restore operation. This overrides all of the sources specified in the *nuget.config* files. Multiple sources can be provided by specifying this option multiple times.
+Specifies a NuGet package source to use during the restore operation. This overrides all of the sources specified in the *nuget.config* files, practically speaking ignoring the nuget.config file. Multiple sources can be provided by specifying this option multiple times.
```
[the remark this answers is not in the document]
Comment (jiimaho) [reply docs/core/tools/dotnet-restore.md]: Okey, then I understand, and I totally agree that the suggested change by me will be misleading. 

I just think the word "overrides" is a bit misleading by itself. When I read it by myself, one interpretation of it was that it overrides sources _with the same name_, but rest stays as-is, which is wrong.

[the remark this answers is not in the document]
Comment (jiimaho) [reply docs/core/tools/dotnet-restore.md]: I did another update to this PR. Please see if you like it better.

**Review by nkolev92 — COMMENTED** [review]
This is semantically correct, so from my side, this change is ok.

**Review by Thraka — APPROVED** [review]

File: docs/core/tools/dotnet-restore.md [hunk 2]
```diff
@@ -171,7 +171,7 @@ Specifies a runtime for the package restore. This is used to restore packages fo
 
 `-s|--source <SOURCE>`
 
-Specifies a NuGet package source to use during the restore operation. This overrides all of the sources specified in the *nuget.config* files. Multiple sources can be provided by specifying this option multiple times.
+Specifies a NuGet package source to use during the restore operation. This overrides all of the sources specified in the *nuget.config* files, effectively reading the NuGet.Config file as if the <packageSource> element was not there. Multiple sources can be provided by specifying this option multiple times.
```
Comment (Thraka) [inline docs/core/tools/dotnet-restore.md]:
```suggestion
Specifies a NuGet package source to use during the restore operation. This overrides all of the sources specified in the *nuget.config* files, effectively reading the *nuget.config* file as if the <packageSource> element was not there. Multiple sources can be provided by specifying this option multiple times.
```

**Review by Thraka — CHANGES_REQUESTED** [review]
@jiimaho  One minor thing that needs to be fixed

**Review by Thraka — APPROVED** [review]
Thanks!

**Review by jiimaho — COMMENTED** [review]
Looks good

**Pull request merged**